test/consistent-destructuring.js

```diff
@@ -0,0 +1,390 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import {outdent} from 'outdent';
+import rule from '../rules/consistent-destructuring';
+
+const ruleTester = avaRuleTester(test, {
+	parserOptions: {
+		ecmaVersion: 2020
+	}
+});
+
+const invalidTestCase = ({code, suggestions}) => {
+	if (!suggestions) {
+		return {
+			code,
+			output: code,
+			errors: [{
+				ruleId: 'consistent-destructuring',
+				messageId: 'consistentDestructuring'
+			}]
+		};
+	}
+
+	return {
+		code,
+		output: code,
+		errors: suggestions.map(suggestion => ({
+			ruleId: 'consistent-destructuring',
+			messageId: 'consistentDestructuring',
+			suggestions: [{
+				messageId: 'consistentDestructuringSuggest',
+				output: suggestion
+			}]
+		}))
+	};
+};
+
+ruleTester.run('consistent-destructuring', rule, {
+	valid: [
```

Comment:
Should we also test that function calls and other complex expressions are not reported by this rule?

For example, this

```js
const { foo } = array.pop();
const { bar } = array.pop();
```

cannot be fixed to this

```js
const { foo, bar } = array.pop();
```

Reply:
^ @medusalix

Reply:
@futpib This rule only reports member expressions on already destructured objects, so your example will be fine in any case. The selector ignores any member expressions on function calls and there are already test cases for that :)

Reply:
Here is a test I think should be valid, but it is reported:
```js
const { a } = foo().bar;
console.log(a, foo().bar.b);
```